flyteplugins/go/tasks/plugins/k8s/batchscheduler/plugins.go

```diff
@@ -0,0 +1,23 @@
+package batchscheduler
+
+import (
+	"context"
+	"github.com/flyteorg/flyte/flyteplugins/go/tasks/plugins/k8s/batchscheduler/scheduler"
+	"github.com/flyteorg/flyte/flyteplugins/go/tasks/plugins/k8s/batchscheduler/scheduler/yunikorn"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SchedulerManager interface {
+	// Mutate is responsible for mutating the object to be scheduled.
+	// It will add the necessary annotations, labels, etc. to the object.
+	Mutate(ctx context.Context, object client.Object) error
```

Comment:
```suggestion
 // Mutate allows the custom scheduler to modify the given Kubernetes object (e.g., a Pod or
    // a custom resource representing a job) before it is submitted to the Kubernetes API.
    // This may include adding annotations, labels, or modifying fields related to scheduling.
    // The object is typically a Kubernetes Pod or a custom resource.
    // It returns an error if the mutation fails.
	Mutate(ctx context.Context, object client.Object) error
```